modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Sorting.tsx

```diff
@@ -0,0 +1,387 @@
+import DropDown from '@clayui/drop-down';
+import ClayForm from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayModal from '@clayui/modal';
+import {fetch, navigate, openModal, openToast} from 'frontend-js-web';
+import React, {useEffect, useRef, useState} from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP, SORTING} from '../Constants';
+import {FDSViewSectionInterface} from '../FDSView';
+import {FDSViewType} from '../FDSViews';
+import OrderableTable from '../components/OrderableTable';
+import RequiredMark from '../components/RequiredMark';
+
+type FDSSortingType = {
+	id: number;
+	name: string;
+	value: string;
+};
+
+type SortType = {
+	label: string;
+	value: string;
+};
+
+interface AddFDSSortingModalContentInterface {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	onSave: Function;
+}
+
+const SORTING_OPTIONS = [SORTING.ASCENDING, SORTING.DESCENDING];
+
+const AddFDSSortingModalContent = ({
+	closeModal,
+	fdsView,
+	onSave,
+}: AddFDSSortingModalContentInterface) => {
+	const [fields, setFields] = useState<Array<string> | null>(null);
+	const [field, setField] = useState<string | null>(null);
+	const [sorting, setSorting] = useState<SortType>(SORTING_OPTIONS[0]);
+
+	useEffect(() => {
+		const getFields = async () => {
+			const {restApplication, restSchema} = fdsView[
+				OBJECT_RELATIONSHIP.FDS_ENTRY_FDS_VIEW
+			];
+
+			const response = await fetch(`/o${restApplication}/openapi.json`);
+
+			if (!response.ok) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const responseJSON = await response.json();
+
+			const properties =
+				responseJSON?.components?.schemas[restSchema]?.properties;
+
+			if (!properties) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const fieldsArray: Array<string> = [];
+
+			const isObjectSchema =
+				responseJSON.components.schemas[restSchema].xml.name ===
+				'ObjectEntry';
+
+			Object.keys(properties).forEach((propertyKey) => {
+				const propertyValue = properties[propertyKey];
+
+				if (isObjectSchema && !propertyValue.extensions) {
+					return;
+				}
+
+				if (propertyKey === 'x-class-name') {
+					return;
+				}
+
+				const type = propertyValue.type;
+
+				if (type === 'object' || type === 'array') {
+					return;
+				}
+
+				fieldsArray.push(propertyKey);
+			});
+
+			setFields(fieldsArray);
+		};
+
+		getFields();
+	}, [fdsView]);
+
+	return (
+		<div className="fds-view-fields-modal">
+			<ClayModal.Header>
+				{Liferay.Language.get('new-default-sort')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayForm.Group>
+					<label htmlFor="field">
+						{Liferay.Language.get('field')}
+
+						<RequiredMark />
+					</label>
+
+					<DropDown
```

Comment:
Curious why you used a DropDown instead of just a ClaySelect? It may be same thing under the hood, but since this is a form I think I would expect a select element.

Reply:
> For the select inputs, I used the ClayDropdown component to make it look closer to the mockups, but let me know if I should switch it to use the ClaySelect component instead.

apparently I just didn't read.

I do think a <select> would be more semantic for our case, but thats more just my opinion than knowing anything concrete.

Reply:
I'll go ahead and switch to using a ClaySelect. It does seem more standard that way and we can switch all ClaySelects in the future if we need to fit any design/UX criteria.